modules/rotating-bound-service-keys.adoc

```diff
@@ -0,0 +1,408 @@
+ifdef::rotate-azure[{entra-short},]
+you can rotate the bound service account signer key.
+
+To rotate the key, you delete the existing key on your cluster, which causes the Kubernetes API server to create a new key.
+To reduce authentication failures during this process, you must immediately add the new public key to the existing issuer file. 
+Once the cluster is using the new key for authentication, you can remove any remaining keys.
+
+//Modified version of the disclaimer from enabling Azure WID on an existing cluster, since there are similar concerns:
+[IMPORTANT]
+====
+The process to rotate OIDC bound service account signer keys is disruptive and takes a significant amount of time.
+Some steps are time-sensitive.
+Before proceeding, observe the following considerations:
+
+* Read the following steps and ensure that you understand and accept the time requirement.
+The exact time requirement varies depending on the individual cluster, but it is likely to require at least one hour.
+
+* To reduce the risk of authentication failures, ensure that you understand and prepare for the time-sensitive steps.
+
+* During this process, you must refresh all service accounts and restart all pods on the cluster.
+These actions are disruptive to workloads.
+To mitigate this impact, you can temporarily halt these services and then redeploy them when the cluster is ready.
+====
+
+.Prerequisites
+
+* You have access to the {oc-first} as a user with the `cluster-admin` role.
+//Permissions requirements (per platform, for install and key rotation)
+include::snippets/ccoctl-provider-permissions-requirements.adoc[]
+* You have configured the `ccoctl` utility.
+* Your cluster is in a stable state.
+You can confirm that the cluster is stable by running the following command:
++
+[source,terminal]
+----
+$ oc adm wait-for-stable-cluster --minimum-stable-period=5s
+----
+
+.Procedure
+
+. Configure the following environment variables:
++
+[source,text]
+----
+ifdef::rotate-aws[]
+INFRA_ID=$(oc get infrastructures cluster -o jsonpath='{.status.infrastructureName}')
+CLUSTER_NAME=${INFRA_ID%-*} <1>
+endif::rotate-aws[]
+ifdef::rotate-gcp[]
+CURRENT_ISSUER=$(oc get authentication cluster -o jsonpath='{.spec.serviceAccountIssuer}')
+GCP_BUCKET=$(echo ${CURRENT_ISSUER} | cut -d "/" -f4)
+endif::rotate-gcp[]
+ifdef::rotate-azure[]
+CURRENT_ISSUER=$(oc get authentication cluster -o jsonpath='{.spec.serviceAccountIssuer}')
+AZURE_STORAGE_ACCOUNT=$(echo ${CURRENT_ISSUER} | cut -d "/" -f3 | cut -d "." -f1)
+AZURE_STORAGE_CONTAINER=$(echo ${CURRENT_ISSUER} | cut -d "/" -f4)
+endif::rotate-azure[]
+----
+ifdef::rotate-aws[]
+<1> This value should match the name of the cluster that was specified in the `metadata.name` field of the `install-config.yaml` file during installation.
+endif::rotate-aws[]
++
+[NOTE]
+====
+You might need to specify some of the values if the corresponding resource names are not equal to the values derived from the cluster name.
```

Comment:
I feel like there might be a less confusing way to say this - but maybe not, and it's just my afternoon brain having a hard time with it. :)